File: gaia-sdk-python/gaia_sdk/graphql/request/type/Failure.py
from gaia_sdk.api.VariableRegistry import VariableRegistry


class Failure(list):

    def reason(self):
        self.append(lambda x: "reason")

    def failure_type(self):
        self.append(lambda x: "failureType")

    def exit_code(self):
        self.append(lambda x: "exitCode")

    def affected_container(self):
        self.append(lambda x: "affectedContainer")

    def logs(self):
        self.append(lambda x: "logs")

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))

File: gaia-sdk-javascript/src/graphql/request/type/Failure.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

export class Failure extends Array<(_:VariableRegistry) => string> {
public _typeName = "Failure";
    public reason = () => { 
        this.push(_ => "reason")
    };

    public failureType = () => { 
        this.push(_ => "failureType")
    };

    public exitCode = () => { 
        this.push(_ => "exitCode")
    };

    public affectedContainer = () => { 
        this.push(_ => "affectedContainer")
    };

    public logs = () => { 
        this.push(_ => "logs")
    };

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

